contributor-book/philosophy_0_80.md

```diff
@@ -0,0 +1,182 @@
+# Nushell design philosophy
+
+## Core philosophy
+
+Nushell is "A shell-first scripting language for working with structured data flowing through pipelines".
+
+Designs should work towards serving this goal. Those that don't should be removed or moved to optional add-ons.
+
+## Core design
+
+### Shell-first
+
+Nushell should work to serve its roll as a shell and a language with a focus on shells. Shell types of activities which include, but aren't limited to:
+
+* Running commands
+* Redirecting stdout/stdin/stderr
+* Unix-only: Properly handling signals
+    * Ctrl-C
+    * Ctrl-D
+    * and others
+* Unix-only: Handling background tasks
+    * Ctrl-Z
+    * Background/foregrounding of tasks
```

Comment:
I'm generally agree with @dandavison

Additionally, it seems the only way to manage task is spawning task by multi-processing, or we have no way to cancel task(we can't cancel a thread).  And it requires our whole engine-state to be serializable/deserializable.  Currently I found `pueue` and `terminal multiplexers` are working really fine to me.   For Ctrl-Z, sometimes I'm using this, but I don't think it's necessary either.

Reply:
This is probably the thing I miss most from bash, but I've found work arounds. Ctrl-z is great for backgrounding neovim and resuming without having to restart the LSP server, which for rust-analyzer has an annoyingly long startup time.

I use kitty these days for a quick new shell session, and there are neovim terminal plugins but the downside is they all take you out to a different window or force you into a smaller screen area

Reply:
> I'm generally agree with @dandavison
> 
> Additionally, it seems the only way to manage task is spawning task by multi-processing, or we have no way to cancel task(we can't cancel a thread). And it requires our whole engine-state to be serializable/deserializable. Currently I found `pueue` and `terminal multiplexers` are working really fine to me. For Ctrl-Z, sometimes I'm using this, but I don't think it's necessary either.

My only gripe with pueue so far is interactivity. Having to send commands through pueue instead of just reattaching a shell to the process is unergonomic

Reply:
I'll turn this into a more general "support for background tasks" as we'll probably want some solution that works reasonably well for them and hopefully one that could work crossplatform.